crates/transaction-pool/src/pool/mod.rs

```diff
@@ -684,6 +685,25 @@ where
         removed
     }
 
+    /// Prune known senders which do not exist in any subpool
+    pub(crate) fn prune_sender_info(&self) {
+        let mut pool = self.pool.write();
+        let identifiers = &self.identifiers;
+
+        for sender_address in self.unique_senders().iter() {
+            if identifiers.read().sender_id(sender_address).is_none() {
+                // will create sender identifier
+                let sender_id = self.get_sender_id(*sender_address);
+
+                pool.remove_sender_info(&sender_id);
+
+                // clean up created sender identifier
+                identifiers.write().remove_sender_address(&sender_id);
+                identifiers.write().remove_sender_id(sender_address);
```

Comment:
we need to be a bit careful here because there are a few race conditions possible.

this is a bit tricky to solve:
we first need to lock the identifiers, then lock the pool and get the unique identifiers
then cleanup the identifiers map

however there's still the possibility that we dont get all of the unique identifiers that we have issued from the pool,

for example:
a new id was issued before we locked identifiers and then we lock the pool before that new id makes its way into the pool.

not sure yet how to fix this